test/Microsoft.IdentityModel.Tokens.Tests/TamperedTokenTests.cs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.JsonWebTokens;
+using Microsoft.IdentityModel.TestUtils;
+using Xunit;
+
+using JwtRegisteredClaimNames = Microsoft.IdentityModel.JsonWebTokens.JwtRegisteredClaimNames;
+
+namespace Microsoft.IdentityModel.Tokens.Tests
+{
+    public class TamperedTokenTests
+    {
+        [Theory, MemberData(nameof(JwtSignatureTruncationTheoryData))]
+        public async Task JwtSignatureTruncation(ValidateTokenTheoryData theoryData)
```

Comment:
>JwtSignatureTruncation [](http://example.com/codeflow?start=26&length=22)

Looks like we're doing some similar testing in SignatureProviderTests::SignatureTrunctation. What's the difference? Looks like that also did symmetric hmac.

Can you help me understand why that didn't catch the issue? Should that test case be updated in any way for protection on the RSA/ECDSA cases?